discovery/gossiper.go

```diff
@@ -1950,16 +1974,28 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 					return nil
 				}
 			} else {
-				// If it's not, we'll only allow a single update
-				// for this channel per block.
+				// If it's not, we'll allow an update per minute
+				// with a maximum burst of 10. If we haven't
+				// seen an update for this channel before, we'll
+				// need to initialize a rate limiter for each
+				// direction.
 				d.Lock()
-				lastUpdateHeight := d.heightForLastChanUpdate[shortChanID]
-				if lastUpdateHeight[direction] == d.bestHeight {
-					log.Debugf("Ignoring update for "+
-						"channel %v due to previous "+
-						"update occurring within the "+
-						"same block %v", shortChanID,
-						d.bestHeight)
+				rateLimiters, ok := d.chanUpdateRateLimiter[shortChanID]
+				if !ok {
+					r := rate.Every(d.cfg.ChannelUpdateInterval)
+					b := d.cfg.MaxChannelUpdateBurst
+					rateLimiters = [2]*rate.Limiter{
+						rate.NewLimiter(r, b),
+						rate.NewLimiter(r, b),
+					}
+					d.chanUpdateRateLimiter[shortChanID] = rateLimiters
+				}
```

Comment:
Do we still need to hold the mutex at this point? AFAICT, we have a pointer here, so we can release and then continue to make modifications to the underlying rate limiter instances.

Reply:
We don't, the rate limiter is safe for concurrent access. Fixed.